### DISCUSSION TITLE
Elastic Search logs /nsm/elasticsearch hot and cold storage

### DISCUSSION THREAD

**Discussion by linsecnet** [Security-Onion-Solutions/securityonion#5544] [Unsupported Versions] [answered]
Hi Community,

We deployed the below security onion architecture in our AWS environment, and I have a question or looking for guidance regarding elastic search storage.

This is what we have:
1 - Manager node - SOC 
3 - Forward Nodes - this is located in 3 different network environments
1 - Fleet Node- this was deployed for OSQuery 
1  -Heavy Node - this was deployed for hot storage
1 - Search Node - this was deployed for a cold storage

What we would like to do is save the logs to the Heavy Node and keep 30 days worth of data ( This is by default I believe)
and after 30 days we would like to move the logs to the search node for cold storage versus being deleted after 30 days.

I searched all over with no luck.

I reviewed this portion of the security documents: https://docs.securityonion.net/en/2.3/elasticsearch.html?highlight=elasticsearch

I modified the /opt/so/saltstack/local/pillar/global.sls

elastic:
  features: False
elasticsearch:
  replicas: 0
  true_cluster: **False**
  true_cluster_name: 'so'
  discovery_nodes: 1
  hot_warm_enabled: **False**

and changed the above from "False" to "True" 
and the only results have changed the status from connected to not connected in the remote clusters in                       
  HTTPS:{IP}kibana/app/management/data/remote_clusters

Any information that anyone has on how to move the log data to a specific node after an X amount of time or instead of having the data moved from one member of the cluster to another would be highly appreciated.

**Comment by linsecnet** [comment]
Since this post, we changed the heavy node for another search node and we upgraded to SO 2.3.8 and still researching.

Also, we changed from hot warm cold to just hot and warm and I believe version 2.3.8 has a hot warm feature.

**Reply by usmanaa10** [reply]
Hi @linsecnet 
             Can you share steps or proper tutorial to deploy Security Onion with multiple nodes in AWS like you did, and configuration to collect logs from VPC, WAF etc. 
I will be thankful to you

**Reply by linsecnet** [reply]
Hi @usmanaa10 
We have the SOC, forward node, and search node in a "management" VPC. Since this post, we did have several forward nodes and we determined that we needed to reduce the number of forward nodes to one major one in our "management" VPC.
Before we reduced the number of forward nodes, we did have peering connections between VPC environments. i.e.

Test VPC  <--- > Management VPC
Production PC < ---> Managment VPC

This allowed the forward node to communicate with the SOC ( manager \ salt manager)
WAZUH manager is configured in the forward nodes in each environment to allow the EC2 instances to communicate with the forward node / WAZUH manager. The forward node / WAZUh manager will then forward the logs to the SOC (manager) with no issues

VPC flow logs were configured to be delivered to an S3 bucket with objects that split up the environment
i.e. S3
- flow-logs-production
- flow-logs-test
SQS was configured to monitor those objects when data was delivered and filebeat was configured for AWS. I used this for a resource : [FileBeat AWS Modulel](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-aws.html)
this is an example:
```
`vpcflow:`
    enabled: false
    var.queue_url: https://sqs.myregion.amazonaws.com/123456/myqueue`
```
This method works flawlessly
Keep in mind you will have to grant proper IAM permission to the forward node and proper permissions on the S3 buckets. Most importantly do not forget the S3 event notification for the SQS.

Although VPC flow logs are configured we have not successfully configured WAF logs. We have not found the proper documentation for WAF logs into Security Onion.

I hope this helps. 

Note: EBS flow logs are region-based so that will be a little trickier if you are configuring those logs.

**Comment by dougburks** [comment, marked as answer]
Hot/warm requires Elasticsearch clustering rather than cross cluster search:
https://docs.securityonion.net/en/2.3/elasticsearch.html#elastic-clustering
